src/TreeNode.js

```diff
@@ -26,7 +26,12 @@ export default class TreeNode {
     this.splitFunction = options.splitFunction;
     this.minNumSamples = options.minNumSamples;
     this.maxDepth = options.maxDepth;
-    this.gainThreshold = options.gainThreshold;
+
+    if (!options.gainThreshold) {
+      this.gainThreshold = 0;
+    } else {
+      this.gainThreshold = options.gainThreshold;
+    }
```

Comment:
This does the same purpose

```suggestion
  this.gainThreshold = options.gainThreshold || 0;
```

Reply:
The problem appears in the random-forest module. Currently the test : RegressionTests.js in that repo fails, the precision currently returned is 0.16. This is because the tree is not built correctly by the RandomForestRegression. When one digs in further, we notice that in the train method, since the variable this.gainThreshold is undefined, we don't enter into the if loop, and don't create left and right subnodes. In this manner the current RandomForestRegression trees all have one level of decision, not multiple levels. This explains why for a specific testing set, all predicted values are the same. This was noted upon several times in the open issues under the random-forest repo.

Reply:
I have made a pull request for the random-forest repo too, but it is currently failing because the changes to this decision-tree-cart repo have not been validated yet. I will see if I can provide a test to show the behavior described in my previous comment.